client/dashboard/sites/overview/test/storage-warning-banner.tsx

```diff
@@ -0,0 +1,113 @@
+/**
+ * @jest-environment jsdom
+ */
+import { screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import nock from 'nock';
+import { render } from '../../../test-utils';
+import { StorageWarningBanner } from '../storage-warning-banner';
+
+interface TestOptions {
+	storage_used_bytes: number;
+	max_storage_bytes: number;
+	isDismissed?: boolean | string;
+}
+
+const siteId = 123;
+
+// Only mocks site and settings fields that are necessary for the tests.
+// Feel free to add more fields as they are needed.
+function mockTestEndpoints( options: TestOptions ) {
+	const { storage_used_bytes, max_storage_bytes, isDismissed = false } = options;
+
+	const mediaStorage = {
+		storage_used_bytes,
+		max_storage_bytes,
+		max_storage_bytes_from_addons: 0,
+	};
+
+	const preferences = {
+		[ `hosting-dashboard-overview-storage-notice-dismissed-${ siteId }` ]:
+			typeof isDismissed === 'string' ? isDismissed : isDismissed && 'today',
+	};
+
+	const scope = nock( 'https://public-api.wordpress.com' )
+		.get( `/rest/v1.1/sites/${ siteId }/media-storage` )
+		.query( true )
+		.reply( 200, mediaStorage )
+		.get( '/rest/v1.1/me/preferences' )
+		.query( true )
+		.reply( 200, { calypso_preferences: preferences } )
+		.post( '/rest/v1.1/me/preferences' )
+		.reply( 200, ( _uri, body ) => body );
+
+	return { mediaStorage, scope };
+}
+
+function renderTestBanner() {
+	// The only site field used by the banner is ID.
+	const testSite = { ID: siteId };
```

Comment:
I usually just use `const testSite = { ID: siteId } as Site;` instead of `any` below; I think it should also work fine.